command_cursor.c

```diff
@@ -115,13 +115,19 @@ int php_mongo_enforce_batch_size_on_command(zval *command, int size TSRMLS_DC)
 		add_assoc_zval(command, "cursor", tmp_cursor);
 	}
 
-	if (Z_TYPE_PP(zcursor) != IS_ARRAY) {
-		php_mongo_cursor_throw(mongo_ce_CursorException, NULL, 32 TSRMLS_CC, "The cursor command's 'cursor' element is not an array");
+	if (Z_TYPE_PP(zcursor) != IS_ARRAY && Z_TYPE_PP(zcursor) != IS_OBJECT) {
+		php_mongo_cursor_throw(mongo_ce_CursorException, NULL, 32 TSRMLS_CC, "The cursor command's 'cursor' element is not an array or object");
 		return 0;
 	}
 
 	if (zend_hash_find(HASH_PP(zcursor), "batchSize", 10, (void**) &zbatchsize) == FAILURE) {
-		add_assoc_long(*zcursor, "batchSize", size);
+		zval *tmp_batchsize;
+
+		MAKE_STD_ZVAL(tmp_batchsize);
+		ZVAL_LONG(tmp_batchsize, size);
+
+		/* Use hash functions here, since "cursor" option may be an object */
```

Comment:
There is no need to use hash functions, as add_assoc_long expands to add_assoc_long_ex, which does:

```
return zend_symtable_update(Z_ARRVAL_P(arg), key, key_len, (void *) &tmp, sizeof(zval *), NULL);
```

and symtable_update is just:

```
    ZEND_HANDLE_NUMERIC(arKey, nKeyLength, zend_hash_index_update(ht, idx, pData, nDataSize, pDest));
    return zend_hash_update(ht, arKey, nKeyLength, pData, nDataSize, pDest);
```